src/DotLiquid.Tests/ConditionTests.cs

```diff
@@ -128,6 +165,112 @@ public void TestContainsWorksOnLongArrays()
+            _context["camry"] = new Car() { Make = "Toyota", Model = "Camry" };
+
+            AssertEvaluatesTrue(left: "array", op: "contains", right: "first");
+            AssertEvaluatesTrue(left: "array", op: "startsWith", right: "first");
+            AssertEvaluatesTrue(left: "array", op: "contains", right: "clone");
+            AssertEvaluatesTrue(left: "array", op: "startsWith", right: "clone");
+            AssertEvaluatesTrue(left: "array", op: "endsWith", right: "last");
+            AssertEvaluatesFalse(left: "array", op: "contains", right: "camry");
+        }
+
+        [Test]
+        public void TestTruthyArray()
+        {
+            _context = new Context(CultureInfo.InvariantCulture);
+            var _array = new List<bool>() { true };
+            _context["array"] = _array.ToArray();
+            _context["first"] = _array.First();
+
+            AssertEvaluatesTrue(left: "array", op: "contains", right: "first");
+            AssertEvaluatesTrue(left: "array", op: "startsWith", right: "first");
+            AssertEvaluatesTrue(left: "array", op: "startsWith", right: "'true'");
+
+            AssertEvaluatesFalse(left: "array", op: "contains", right: "'true'"); // to be re-evaluated in #362
+        }
+
+        [Test]
+        public void TestCharArrays()
+        {
+            _context = new Context(CultureInfo.InvariantCulture);
+            var _array = new List<char> { 'A', 'B', 'C' };
+            _context["array"] = _array.ToArray();
+            _context["first"] = _array.First();
+            _context["last"] = _array.Last();
+
+            AssertEvaluatesTrue(left: "array", op: "contains", right: "'A'");
+            AssertEvaluatesTrue(left: "array", op: "contains", right: "first");
+            AssertEvaluatesTrue(left: "array", op: "startsWith", right: "first");
+            AssertEvaluatesFalse(left: "array", op: "contains", right: "'a'");
+            AssertEvaluatesFalse(left: "array", op: "contains", right: "'X'");
+            AssertEvaluatesTrue(left: "array", op: "contains", right: "'B'");
+            AssertEvaluatesTrue(left: "array", op: "contains", right: "'C'");
+            AssertEvaluatesTrue(left: "array", op: "endsWith", right: "last");
+        }
+
+        [Test]
+        public void TestByteArrays()
+        {
+            _context = new Context(CultureInfo.InvariantCulture);
+            var _array = new List<byte> { 0x01, 0x02, 0x03, 0x30 };
+            _context["array"] = _array.ToArray();
+            _context["first"] = _array.First();
+            _context["last"] = _array.Last();
+
+            AssertEvaluatesFalse(left: "array", op: "contains", right: "0");
+            AssertEvaluatesFalse(left: "array", op: "contains", right: "'0'");
+            AssertEvaluatesTrue(left: "array", op: "startsWith", right: "first");
+            AssertEvaluatesTrue(left: "array", op: "contains", right: "first");
+            AssertEvaluatesFalse(left: "array", op: "contains", right: "1");

```

Comment:
Was it possible to get a match with a byte at all? Is there a way to write the right operand as a byte?